### DISCUSSION TITLE
Broke my server upgrading to 3.8.0

### DISCUSSION THREAD

**Discussion by NickVeit** [jamulussoftware/jamulus#1817] [Technical Support & Help] [answered]
I know little about Ubuntu (20.04) but I have been running a server for ages and have done several upgrades. Now moving to 3.8.0 I now have a problem...

sudo make
g++ -Wl,--as-needed -Wl,-O1 -o Jamulus x86cpu.o x86_celt_map.o pitch_sse.o pitch_sse2.o vq_sse2.o celt_lpc_sse4_1.o
 pitch_sse4_1.o NSQ_sse4_1.o NSQ_del_dec_sse4_1.o x86_silk_map.o VAD_sse4_1.o VQ_WMat_EC_sse4_1.o sound.o buffer.o 
channel.o client.o main.o protocol.o jamcontroller.o server.o serverlist.o serverlogging.o settings.o signalhandler
.o socket.o soundbase.o util.o jamrecorder.o creaperproject.o cwavestream.o audiomixerboard.o chatdlg.o clientsetti
ngsdlg.o connectdlg.o clientdlg.o serverdlg.o multicolorled.o levelmeter.o analyzerconsole.o bands.o celt.o celt_de
coder.o celt_encoder.o celt_lpc.o cwrs.o entcode.o entdec.o entenc.o kiss_fft.o laplace.o mathops.o mdct.o modes.o 
pitch.o quant_bands.o rate.o vq.o A2NLSF.o ana_filt_bank_1.o biquad_alt.o bwexpander.o bwexpander_32.o check_contro
l_input.o CNG.o code_signs.o control_audio_bandwidth.o control_codec.o control_SNR.o debug.o decoder_set_fs.o decod
e_core.o decode_frame.o decode_indices.o decode_parameters.o decode_pitch.o decode_pulses.o dec_API.o encode_indice
s.o encode_pulses.o enc_API.o apply_sine_window_FLP.o autocorrelation_FLP.o burg_modified_FLP.o bwexpander_FLP.o co
rrMatrix_FLP.o encode_frame_FLP.o energy_FLP.o find_LPC_FLP.o find_LTP_FLP.o find_pitch_lags_FLP.o find_pred_coefs_
FLP.o inner_product_FLP.o k2a_FLP.o LPC_analysis_filter_FLP.o LTP_analysis_filter_FLP.o LTP_scale_ctrl_FLP.o noise_
shape_analysis_FLP.o pitch_analysis_core_FLP.o process_gains_FLP.o residual_energy_FLP.o scale_copy_vector_FLP.o sc
ale_vector_FLP.o schur_FLP.o sort_FLP.o warped_autocorrelation_FLP.o wrappers_FLP.o gain_quant.o HP_variable_cutoff
.o init_decoder.o init_encoder.o inner_prod_aligned.o interpolate.o lin2log.o log2lin.o LPC_analysis_filter.o LPC_f
it.o LPC_inv_pred_gain.o LP_variable_cutoff.o NLSF2A.o NLSF_decode.o NLSF_del_dec_quant.o NLSF_encode.o NLSF_stabil
ize.o NLSF_unpack.o NLSF_VQ.o NLSF_VQ_weights_laroia.o NSQ.o NSQ_del_dec.o pitch_est_tables.o PLC.o process_NLSFs.o
 quant_LTP_gains.o resampler.o resampler_down2.o resampler_down2_3.o resampler_private_AR2.o resampler_private_down
_FIR.o resampler_private_IIR_FIR.o resampler_private_up2_HQ.o resampler_rom.o shell_coder.o sigm_Q15.o sort.o stere
o_decode_pred.o stereo_encode_pred.o stereo_find_predictor.o stereo_LR_to_MS.o stereo_MS_to_LR.o stereo_quant_pred.
o sum_sqr_shift.o tables_gain.o tables_LTP.o tables_NLSF_CB_NB_MB.o tables_NLSF_CB_WB.o tables_other.o tables_pitch
_lag.o tables_pulses_per_block.o table_LSF_cos.o VAD.o VQ_WMat_EC.o analysis.o mlp.o mlp_data.o opus.o opus_decoder
.o opus_encoder.o repacketizer.o qrc_resources.o moc_sound.o moc_channel.o moc_client.o moc_protocol.o moc_jamcontr
oller.o moc_server.o moc_serverlist.o moc_settings.o moc_socket.o moc_soundbase.o moc_testbench.o moc_util.o moc_ja
mrecorder.o moc_creaperproject.o moc_signalhandler.o moc_audiomixerboard.o moc_chatdlg.o moc_clientsettingsdlg.o mo
c_connectdlg.o moc_clientdlg.o moc_serverdlg.o moc_levelmeter.o moc_analyzerconsole.o moc_multicolorled.o   -lpthre
ad /usr/lib/x86_64-linux-gnu/libQt5Widgets.so /usr/lib/x86_64-linux-gnu/libQt5Gui.so /usr/lib/x86_64-linux-gnu/libQ
t5Network.so /usr/lib/x86_64-linux-gnu/libQt5Xml.so /usr/lib/x86_64-linux-gnu/libQt5Concurrent.so /usr/lib/x86_64-l
inux-gnu/libQt5Core.so /usr/lib/x86_64-linux-gnu/libGL.so   
/usr/bin/ld: NSQ_sse4_1.o: in function `silk_NSQ_sse4_1':
NSQ_sse4_1.c:(.text+0x11a9): undefined reference to `silk_noise_shape_quantizer'
/usr/bin/ld: VAD_sse4_1.o: in function `silk_VAD_GetSA_Q8_sse4_1':
VAD_sse4_1.c:(.text+0x35a): undefined reference to `silk_VAD_GetNoiseLevels'
collect2: error: ld returned 1 exit status
make: *** [Makefile:1123: Jamulus] Error 1

**Comment by npostavs** [comment, marked as answer]
Try running `make clean` first.

**Comment by pljones** [comment]
As we distribute pre-compiled binaries for Ubuntu, it might be better if you used the server .deb file.  It would possibly mean reinstalling your server (following one of the new guides based around the .deb file) - but upgrading becomes very simple then.

**Comment by NickVeit** [comment]
Many thanks. "sudo make clean" before the other "make" commands fixed the problem and I'm up and running. 

My upgrade process has been (and will hopefully continue to be) very quick and simple but I'll consider deb files (and a reinstallation) if need be.

Very grateful guys!

**Reply by npostavs** [reply]
By the way, usual practice is only to use `sudo` for the `make install` step.

**Reply by NickVeit** [reply]
Thanks, I simply used sudo as sometimes have to on other commands...I appreciate your advice...I know very little about Unix, my use of it being out of necessity.

**Reply by pljones** [reply]
Using `sudo` without understanding why is handing over your doorkeys to someone you don't know, basically.

**Reply by ann0see** [reply]
Everyone starts at zero first ;-). I suggest you to learn some Linux cli commands which will make your life easier and help you understand what you’re doing. 

You should also ensure that your server is reasonably protected against common attacks (brute force attacks on your SSH login,…). I suggest you to read an article like this one: https://www.informaticar.net/security-hardening-ubuntu-20-04/ and think about fail2ban, ssh authentication with keys (which is not only more secure, but also much more convenient since there’s no need to type a login password again).

**Comment by NickVeit** [comment]
Thanks for the responses. My rather standard and simple server (Jumulus plus DNS update client) is hosted by Google where there is always a backup image. It is only switched on for private rehearsals. I retired from a career in technology 20+ years ago and have no interest in learning another operating system in addition being the one who fixes technical and organisational things all the time :). Much of my interest is now in preparing our audio gear and performers for live events, and enjoying regular evening jams with new friends thanks to the magic of Jamulus etc. for which we are all very grateful.
